### ISSUE TITLE
`SelectObjectContent` `ScanRange` does not accept zero-start

### ISSUE THREAD

**Issue by Licht-T** [awslabs/aws-sdk-rust#630]
### Describe the bug

[ScanRange is the zero-start range](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ScanRange.html#AmazonS3-Type-ScanRange-Start), but Rust SDK doesn't accept zero as `Start` value.

https://github.com/awslabs/aws-sdk-rust/blob/b06cd25e2ade91951d4ce52ced908c49ac280401/sdk/s3/src/xml_ser.rs#L1114-L1121

### Expected Behavior

When the range is [0, 54], the `ScanRange` element of request body should be:
```
<ScanRange><Start>0</Start><End>54</End></ScanRange>
```

### Current Behavior

When the range is [0, 54], it is currently:
```
<ScanRange><End>54</End></ScanRange>
```

### Reproduction Steps

```rust
                    let mut select_object_content_req = client
                        .select_object_content()
                        .bucket(bucket.to_owned())
                        .key(key.to_owned())
(snip)
                        .scan_range(
                            ScanRange::builder()
                                .start(0)
                                .end(54)
                                .build(),
                        )
```

### Version

```text
v0.49.0
```

### Environment details (OS name and version, etc.)

Ubuntu 22.04

**Comment by Velfi** [comment]
@Licht-T Thanks for reporting this. I'll take a look today and see if I can fix it.

**Comment by Velfi** [comment]
I was able to reproduce this issue. It should be solvable by customizing the serializer to serialize zeros for that operation. Working on that now.

**Comment by Velfi** [comment]
I was able to fix this with a [serializer codegen update](https://github.com/awslabs/smithy-rs/pull/1830). I'll update this post again once the fix is released (should be less than 2 weeks.)

**Comment by Licht-T** [comment]
Thanks, @Velfi. Lookin forward to the PR merged.

**Comment by Velfi** [comment]
It looks like this is going to take longer than I expected. I'm going to be working this week to find a better solution to this issue.

**Comment by Velfi** [comment]
Hey @Licht-T, I'm sorry to say that there's been no new developments. My current stance is that I should probably close the current PR and create a new fix that solves this issue with input tracking, per @jdisanti 's [comment](https://github.com/awslabs/smithy-rs/pull/1830#issuecomment-1273836959).

**Comment by Velfi** [comment]
Sorry again for the lack of movement on this. This week I'm going to check if a recent PR that changes how we handle values like `StartRange` and `EndRange` fixes this.

**Comment by Velfi** [comment]
After looking into this, I believe it's going to be blocked until we implement smithy IDL v2 support. This is due to the fact that we need to interact with the `default` trait which is only available in IDL v2.

**Comment by Velfi** [comment]
Blocked on https://github.com/awslabs/aws-sdk-rust/issues/536 and https://github.com/awslabs/smithy-rs/issues/1767

**Comment by rcoh** [comment]
This **should** be fixed when S3 adds `@clientOptional` to all primitives.

**Comment by rcoh** [comment]
This should be fixed in v0.38, we are verifying

**Comment by jdisanti** [comment]
Confirmed this is fixed in the 1.x SDK version.
